### PR TITLE
refractor(storage): wait_until_queue_empty -> wait

### PR DESCRIPTION
**Pull request by tartavull**
refractor(tasks): wait_until_queue_empty -> wait
To speed up file uploading and downloading Storage implements
asyncronous operations. When subsequent code depends on the
operations being performed wait() is required. wait() blocks
until all operations have being executed.

It renames wait_until_queue_empty() to the less verbose wait()
in the Storage class.

This commit also adds wait() to task which required it and were
neglecting it.

### CODE REVIEW

File: python/neuroglancer/pipeline/precomputed.py [hunk 1]
```diff
@@ -72,10 +72,10 @@ def __getitem__(self, slices):
         return_volume = np.empty(
             shape=self._get_slices_shape(aligned_slices),
             dtype=self.info['data_type'])
-
+        
+        self._storage.wait()
```
Comment (tartavull) [inline python/neuroglancer/pipeline/precomputed.py]: Why is this necessary?

File: python/neuroglancer/pipeline/tasks.py [hunk 2]
```diff
@@ -317,6 +320,7 @@ def _download_input_chunk(self):
                 self._storage.put_file(
                     file_path='{}/{}:{}'.format(self._info['mesh'],last_id, self.lod),
                     content=json.dumps({"fragments": last_fragments}))
+                self._storage.wait()
```
Comment (tartavull) [inline python/neuroglancer/pipeline/tasks.py]: I don't think this is necessary either, is it?

Comment (tartavull) [inline python/neuroglancer/pipeline/tasks.py]: or at least could be outside the for loop to benefit from async performance

File: python/neuroglancer/pipeline/precomputed.py (hunk 1 above)
Comment (william-silversmith) [inline python/neuroglancer/pipeline/precomputed.py]: ima b honest i have no idea where that came from.

File: python/neuroglancer/pipeline/tasks.py (hunk 2 above)
Comment (william-silversmith) [inline python/neuroglancer/pipeline/tasks.py]: In another branch it was :-/

File: python/neuroglancer/pipeline/precomputed.py (hunk 1 above)
Comment (tartavull) [inline python/neuroglancer/pipeline/precomputed.py]: Now that we broke apart this PR, it's easier to catch stuff like this.

File: python/neuroglancer/pipeline/storage.py [hunk 3]
```diff
@@ -204,6 +205,8 @@ class FileInterface(object):
 
     def __init__(self, path):
         self._path = path
+        self.dir_creation = Lock()
```
Comment (william-silversmith) [inline python/neuroglancer/pipeline/storage.py]: Using locks might be a good idea versus recovering from errors, but testing demonstrates random deadlocks on `test/test_storage.py::test_list`

Another reason this may not be the right implementation is that each interface thread creating and acquiring a lock to work with means that the lock does not synchronize between threads. The threads must share a common lock for it to be meaningful.

Comment (tartavull) [inline python/neuroglancer/pipeline/storage.py]: good catch, I made it a class method.

**Comment by william-silversmith** [comment]
That will work but will make threaded file fetching the same as single
threaded. File level locks are needed and the OS already provides them. I
would advise using the subsequent PR instead.
